README.md

```diff
@@ -2,17 +2,80 @@
 
 [![GoDoc](https://godoc.org/github.com/go-bdd/gobdd?status.svg)](https://godoc.org/github.com/go-bdd/gobdd) [![Coverage Status](https://coveralls.io/repos/github/go-bdd/gobdd/badge.svg?branch=master)](https://coveralls.io/github/go-bdd/gobdd?branch=master)
 
-This is a BDD testing framework. Uses gherkin for the test's syntax. Project is under active development.
+This is a BDD testing framework. Uses gherkin for the test's syntax. From version 1.0, the API is stable.
 
-More detailed documentation can be found on the docs page: https://go-bdd.github.io/gobdd/
+## Why did I make the library?
+There is godog library for BDD tests in Go. I found this library useful but it run as an external application which compiles our code. It has a several disadvantages:
+
+ * no debugging (breakpoints) in the test. Sometimes it’s useful to go through the whole execution step by step
+ * metrics don’t count the test run this way
+ * some style checkers recognise tests as dead code
+ * it’s impossible to use built-in features like build constraints.
+ * no context in steps - so the state have to be stored somewhere else - in my opinion, it makes the maintenance harder
+
+## Quick start
+
+Add the package to your project:
+
+```
+go get github.com/go-bdd/gobdd
+```
+
+Add a new test `main_test.go`:
+
+```go
+func add(t gobdd.StepTest, ctx context.Context, var1, var2 int) context.Context{
+	res := var1 + var2
+	ctx.Set("sumRes", res)
+	return ctx
+}
+
+func check(t gobdd.StepTest, ctx context.Context, sum int) context.Context{
+	received, err := ctx.GetInt("sumRes")
+	if err != nil {
+		t.Error(err)
+		return ctx
+	}
+
+	if sum != received {
+		t.Error(errors.New("the math does not work for you"))
+	}
+
+	return ctx
+}
+
+func TestScenarios(t *testing.T) {
+	suite := NewSuite(t)
+	suite.AddStep(`I add (\d+) and (\d+)`, add)
+	suite.AddStep(`I the result should equal (\d+)`, check)
+	suite.Run()
+}
+```
+
+Inside `features` folder create your scenarios. Here is an example:
```

Comment:
Would it make sense to move this section related to features above the section related to test steps? I usually write (or already have) feature files and only then go on implementing the steps.

Reply:
good point!